cli/src/commands/config.rs

```diff
@@ -296,9 +297,56 @@ pub(crate) fn cmd_config_set(
             path = config_path.display()
         )));
     }
+
+    // If the user is trying to change the author config, we should warn them that
+    // it won't affect the working copy author
+    if args.name == ConfigNamePathBuf::from_iter(vec!["user", "name"]) {
+        let config = command.settings().config();
+        let user_name: String = config.get("user.name")?;
+
+        // only warn if the author is actually changing
+        if user_name != args.value {
+            let user_email: String = config.get("user.email")?;
+            warn_wc_author(&user_name, &user_email, command, ui)?
+        }
+    } else if args.name == ConfigNamePathBuf::from_iter(vec!["user", "email"]) {
+        let config = command.settings().config();
+        let user_email: String = config.get("user.email")?;
+
+        // only warn if the author is actually changing
+        if user_email != args.value {
+            let user_name: String = config.get("user.name")?;
+            warn_wc_author(&user_name, &user_email, command, ui)?
+        }
+    }
     write_config_value_to_file(&args.name, &args.value, &config_path)
 }
 
+fn warn_wc_author(
+    user_name: &str,
+    user_email: &str,
+    command: &CommandHelper,
+    ui: &mut Ui,
+) -> Result<(), CommandError> {
+    let helper = command.workspace_helper(ui)?;
+    let repo = helper.repo();
+    let maybe_wc_commit = helper
+        .get_wc_commit_id()
+        .map(|id| repo.store().get_commit(id))
+        .transpose()?;
+
+    let change_hash = match maybe_wc_commit {
+        Some(commit) => format!("{} ", &short_change_hash(commit.change_id())[..8]), /* extra space for formatting */
+        None => String::from(""),
+    };
+    Ok(writeln!(
+        ui.warning_default(),
+        "This setting will only impact future commits.\nThe author of the working copy \
+         {change_hash}will stay \"{user_name} <{user_email}>\". \nTo change it, use \"jj describe \
+         --reset-author --no-edit\""
+    )?)
```

Comment:
nit: I don't think printing hash is needed (for warning/hint), but maybe you can use `write_commit_summary()` for this purpose.

Reply:
> maybe you can use write_commit_summary() for this purpose.

It prints a lot of different info directly to the formatter so it's of no use for me. I'll remove the reference to the change id.